src/snowflake/connector/pandas_tools.py

```diff
@@ -152,37 +152,27 @@ def write_pandas(
         )
 
     if quote_identifiers:
-        location = (
-            (('"' + database + '".') if database else "")
-            + (('"' + schema + '".') if schema else "")
-            + ('"' + table_name + '"')
+        location = (('"' + database + '".') if database else "") + (
+            ('"' + schema + '".') if schema else ""
         )
     else:
-        location = (
-            (database + "." if database else "")
-            + (schema + "." if schema else "")
-            + (table_name)
+        location = (database + "." if database else "") + (
+            schema + "." if schema else ""
         )
     if chunk_size is None:
         chunk_size = len(df)
+
+    def random_name(length: int = 10) -> str:
+        return "".join(random.choice(string.ascii_lowercase) for _ in range(length))
+
     cursor = conn.cursor()
-    stage_name = None  # Forward declaration
-    while True:
-        try:
-            stage_name = "".join(
-                random.choice(string.ascii_lowercase) for _ in range(5)
-            )
-            create_stage_sql = (
-                "create temporary stage /* Python:snowflake.connector.pandas_tools.write_pandas() */ "
-                '"{stage_name}"'
-            ).format(stage_name=stage_name)
-            logger.debug(f"creating stage with '{create_stage_sql}'")
-            cursor.execute(create_stage_sql, _is_internal=True).fetchall()
-            break
-        except ProgrammingError as pe:
-            if pe.msg.endswith("already exists."):
-                continue
-            raise
+    stage_name = random_name()
+    create_stage_sql = (
+        "create temporary stage /* Python:snowflake.connector.pandas_tools.write_pandas() */ "
+        '"{stage_name}"'
+    ).format(stage_name=stage_name)
+    logger.debug(f"creating stage with '{create_stage_sql}'")
+    cursor.execute(create_stage_sql, _is_internal=True).fetchall()
```

Comment:
If the random name exists then this will just raise an exception now

Reply:
the random name should never conflict because the probability of conflict is (1/26)^10 ~1e-14